AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/StorageEngineTests.swift

```diff
@@ -0,0 +1,156 @@
+            storageAdapter = try SQLiteStorageEngineAdapter(connection: connection)
+            try storageAdapter.setUp(modelSchemas: StorageEngine.systemModelSchemas)
+
+            let syncEngine = try RemoteSyncEngine(storageAdapter: storageAdapter,
+                                                  dataStoreConfiguration: .default)
+            storageEngine = StorageEngine(storageAdapter: storageAdapter,
+                                          dataStoreConfiguration: .default,
+                                          syncEngine: syncEngine,
+                                          validAPIPluginKey: validAPIPluginKey,
+                                          validAuthPluginKey: validAuthPluginKey)
+            ModelRegistry.register(modelType: Post.self)
+            ModelRegistry.register(modelType: Comment.self)
+//            guard let postModelSchema = ModelRegistry.modelSchema(from: Post.) else {
+//                XCTFail("Failed to generate model schema")
+//                return
+//            }
+            do {
+            try storageEngine.setUp(modelSchemas: [Post.schema])
+            try storageEngine.setUp(modelSchemas: [Comment.schema])
+
+            } catch {
+                XCTFail("Failed to setup storage engine")
+            }
+        } catch {
+            XCTFail(String(describing: error))
+            return
+        }
+    }
+
+    func testCreationQueryDelete_NAME_TBD() {
+        let post = Post(title: "post1", content: "content1", createdAt: .now())
+        guard case .success(let savedPost1) = saveModelSynchronous(model: post) else {
+                XCTFail("Failed to save")
+                return
+        }
+
+        let comment1 = Comment(content: "comment1ForPost1", createdAt: .now() + .weeks(1), post: savedPost1)
+        let comment2 = Comment(content: "comment2ForPost1", createdAt: .now() + .days(1), post: savedPost1)
+        guard case .success(let savedComment1) = saveModelSynchronous(model: comment1),
+            case .success(let savedComment2) = saveModelSynchronous(model: comment2) else {
+            XCTFail("Failed to save either comment")
+            return
+        }
+
+        guard case .success(let queriedPost) =
+            querySingleModelSynchronous(modelType: Post.self, predicate: Post.keys.id == savedPost1.id) else {
+            XCTFail("Failed to query for single post")
+            return
+        }
+
+        guard case .success(let comments) =
+            queryModelSynchronous(modelType: Comment.self, predicate: Comment.keys.post == savedPost1.id) else {
+            XCTFail("Failed to query comment")
+            return
+        }
+        XCTAssertEqual(comments.count, 2)
+    }
+
+    /**
+     * Below are synchronous conveinence methods.  Please do not add any calls to XCTFail()
+     * in these conveinence methods.  Failures should be handled in the body of the unit test.
+     */
+    func saveModelSynchronous<M: Model>(model: M) -> DataStoreResult<M> {
```

Comment:
i wonder how well this translate to API results as that results a success/failure and inside the success is a graphQL response with the data or graphql error that has transformation/partial/error cases. This nested response means that the helper for API requests will result in aggregating the results in the helper method and returning `Result<M>` so that the caller can simply do `guard case .success(let m) = ..`